docs/developer-docs/security/security-best-practices/inter-canister-calls.mdx

```diff
@@ -269,11 +279,71 @@ mod test {
 }
 ```
 
+</TabItem>
+
+<TabItem value="motoko" label="Motoko">
+
+```motoko
+import Result "mo:base/Result";
+import Map "mo:base/HashMap";
+import Error "mo:base/Error";
+import Principal "mo:base/Principal";
+
+actor {
+
+  let pending_requests = Map.HashMap<Principal, Bool>(10, Principal.equal, Principal.hash);
```

Comment:
nit: IIUC you use a HashMap because there is no set right? Would it make sense to explain this e.g. by adding this comment?
```
  // We use HashMap (and set a dummy booleans as values) even though a set would be sufficient. However, there is currently no Set implementation in Motoko.
```

Reply:
AFAIK the Motoko base lib does not have a set structure. But I could be wrong. 
@ggreif?

Reply:
Well a set is basically a map `<KeyType, ()>`, i.e. the value type is unit. So It should be easy to typedef it that way.

But I also think that a grant is being prepared to come up with modern implementations of associative containers. So we might have an efficient `Set` someday soon.